pkg/trace/pb/decoder_test.go

```diff
@@ -44,3 +45,107 @@ func TestParseFloat64(t *testing.T) {
 	assert.NoError(err)
 	assert.Equal(3.14, f)
 }
+
```

Comment:
What is happening in case of invalid v0.5 payload received? This is likely to happen in case of a tracer bug.

- We must be sure it doesn't crash the Agent.
- Should we drop just affected spans? full traces? full payloads?
- We must have a very precise metric to track these cases.

Let's also add tests for it.

Reply:
Got it! It makes sense and is good to me.

Concerning "crashes", since we do some more "manual" decoding, I was concerned about cases we would not be catching properly. At the time of the review, `dict[idx]` with `idx < 0` would have crashed (fixed since then).
Let's be sure they are no other case which could crash if ingested data is invalid.